src/main/java/org/jabref/gui/libraryproperties/LibraryPropertiesDialogView.java

```diff
@@ -0,0 +1,214 @@
+
+import javafx.event.ActionEvent;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.CheckBox;
+import javafx.scene.control.ComboBox;
+import javafx.scene.control.RadioButton;
+import javafx.scene.control.TextField;
+import javafx.scene.control.ToggleGroup;
+import javafx.scene.layout.VBox;
+
+import org.jabref.gui.BasePanel;
+import org.jabref.gui.DialogService;
+import org.jabref.gui.SaveOrderConfigDisplay;
+import org.jabref.gui.cleanup.FieldFormatterCleanupsPanel;
+import org.jabref.gui.util.BaseDialog;
+import org.jabref.logic.cleanup.Cleanups;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.database.shared.DatabaseLocation;
+import org.jabref.model.metadata.MetaData;
+import org.jabref.model.metadata.SaveOrderConfig;
+import org.jabref.preferences.PreferencesService;
+
+import com.airhacks.afterburner.views.ViewLoader;
+
+public class LibraryPropertiesDialogView extends BaseDialog<Void> {
+
+    @FXML private VBox contentVbox;
+    @FXML private ComboBox<Charset> encoding;
+    @FXML private TextField generalFileDirectory;
+    @FXML private Button browseGeneralFileDir;
+    @FXML private TextField userSpecificFileDirectory;
+    @FXML private Button browseUserSpefiicFileDir;
+    @FXML private VBox saveOrdervbox;
+    @FXML private ToggleGroup saveOrderToggleGroup;
+    @FXML private RadioButton saveInOriginalOrder;
+    @FXML private RadioButton saveInSpecifiedOrder;
+    @FXML private CheckBox protect;
+    private SaveOrderConfigDisplay saveOrderPanel;
+    private BasePanel panel;
+    private LibraryPropertiesDialogViewModel viewModel;
+    private final DialogService dialogService;
+    @Inject private PreferencesService preferencesService;
+    private FieldFormatterCleanupsPanel fieldFormatterCleanupsPanel;
+    private Object oldSaveOrderConfig;
+
+    public LibraryPropertiesDialogView(BasePanel panel, DialogService dialogService) {
+        this.dialogService = dialogService;
+        this.panel = panel;
+        ViewLoader.view(this)
+                  .load()
+                  .setAsDialogPane(this);
+
+        setResultConverter(btn -> {
+            if (btn == ButtonType.OK) {
+                storeSettings();
+            }
+            return null;
+        });
+
+        setTitle(Localization.lang("Library properties"));
+
+    }
+
+    @FXML
+    private void initialize() {
+        viewModel = new LibraryPropertiesDialogViewModel(panel, dialogService, preferencesService);
+
+        generalFileDirectory.textProperty().bindBidirectional(viewModel.generalFileDirectoryPropertyProperty());
+        userSpecificFileDirectory.textProperty().bindBidirectional(viewModel.userSpecificFileDirectoryProperty());
+
+        encoding.itemsProperty().bind(viewModel.encodingsProperty());
+        encoding.valueProperty().bindBidirectional(viewModel.selectedEncodingProperty());
+        saveOrderPanel = new SaveOrderConfigDisplay();
```

Comment:
Can you please try to declare these panels in the fxml (might be connected to some small changes to these classes but shouldn't be that hard).

Reply:
Any tutorial what I need to do to make it avaiable in the fxml? Probably I need to inherit some javafx container, right?

Reply:
yes, the control needs to derive from `Node` (probably best to derive from an appropriate layout pane).